extensions/kubeapi/namespaces/create.go

```diff
@@ -21,17 +19,20 @@ import (
 
 // CreateNamespace is a helper function that uses the dynamic client to create a namespace on a project.
 // It registers a delete function with a wait.WatchWait to ensure the namspace is deleted cleanly.
-func CreateNamespace(client *rancher.Client, namespaceName, containerDefaultResourceLimit string, labels, annotations map[string]string, project *management.Project) (*coreV1.Namespace, error) {
-	// Namespace object for a project name space
+func CreateNamespace(client *rancher.Client, clusterID, projectName, namespaceName, containerDefaultResourceLimit string, labels, annotations map[string]string) (*coreV1.Namespace, error) {
```

Comment:
can we change this to projectID as its a projectID rather than a name?

Reply:
Initially I had projectID but I later changed it to projectName because there is no project ID in public API. 
```
> k get projects p-bg8wt -n c-m-5w796d5p -o yaml | egrep -h 'id|name'
    lifecycle.cattle.io/create.project-namespace-auth_c-m-5w796d5p: "true"
  - clusterscoped.controller.cattle.io/project-namespace-auth_c-m-5w796d5p
  name: p-bg8wt
  namespace: c-m-5w796d5p
  uid: e839244a-a956-427c-b444-8568d45003a0
> 
```